src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IEnumerableConverter.cs

```diff
@@ -14,14 +15,16 @@ internal sealed class IEnumerableConverter<TCollection>
         : JsonCollectionConverter<TCollection, object?>
         where TCollection : IEnumerable
     {
+        private readonly bool _isDeserializable = typeof(TCollection).IsAssignableFrom(typeof(List<object?>));
```

Comment:
is this type usable when this is false? if not, should we throw in the constructor?

Reply:
It's usable for serialization, it would be a breaking change if we had it throw for the particular collection type.